crate/server/src/core/operations/register.rs

```diff
@@ -0,0 +1,106 @@
+use std::sync::Arc;
+
+use cosmian_kms_server_database::reexport::{
+    cosmian_kmip::{
+        self,
+        kmip_2_1::{
+            kmip_objects::ObjectType,
+            kmip_operations::{Register, RegisterResponse},
+            kmip_types::UniqueIdentifier,
+        },
+    },
+    cosmian_kms_interfaces::SessionParams,
+};
+use tracing::{debug, trace};
+
+use crate::{
+    core::{
+        KMS,
+        operations::import::{
+            process_certificate, process_private_key, process_public_key, process_symmetric_key,
+        },
+        retrieve_object_utils::user_has_permission,
+    },
+    error::KmsError,
+    kms_bail,
+    result::KResult,
+};
+
+pub(crate) async fn register(
+    kms: &KMS,
+    request: Register,
+    owner: &str,
+    params: Option<Arc<dyn SessionParams>>,
+    privileged_users: Option<Vec<String>>,
+) -> KResult<RegisterResponse> {
+    trace!("Register: {}", serde_json::to_string(&request)?);
+    if request.protection_storage_masks.is_some() {
+        kms_bail!(KmsError::UnsupportedPlaceholder)
+    }
+
+    // To register an object, check that the user has `Create` access right
+    if let Some(users) = privileged_users.clone() {
+        let has_permission = user_has_permission(
+            owner,
+            None,
+            &cosmian_kmip::kmip_2_1::KmipOperation::Create,
```

Comment:
We need to comment here that the delegation of Create covers Create Import and Register
The documentation must also be clear on that aspect